sns_ik_examples/src/ik_tests.cpp

```diff
@@ -60,20 +60,45 @@ double getDeltaWithLimits(double value, double desired_delta,
   }
 }
 
-bool in_vel_bounds(KDL::JntArray vals, KDL::JntArray vels)
+bool in_vel_bounds(const KDL::JntArray& vals, const KDL::JntArray& vels)
 {
   for(size_t i; i < vels.data.size(); i++){
       if(vals(i) < -vels(i) || vals(i) > vels(i)){
           return false;
       }
-      /*else{
-          std::cout<<vals(i)<<' ';
-      }*/
   }
-  //std::cout<<std::endl;
   return true;
 }
 
+bool in_pos_bounds(const KDL::JntArray& jnt_values, const KDL::JntArray& jnt_lower,
+                   const KDL::JntArray& jnt_upper)
+{
+  for(size_t i; i < jnt_values.data.size(); i++){
+      if(jnt_values(i) < jnt_lower(i) || jnt_values(i) > jnt_upper(i)){
+          return false;
+      }
+  }
+  return true;
+}
+
+double nullspace_influence(const KDL::JntArray& jnt_result, const KDL::JntArray& jnt_ns_result,
```

Comment:
Instead of counting the number of joints which were closer, a more robust method would be to compare the L2 norm of the difference between result and bias.

error = (jnt_result - jnt_ns_bias).Norm()